identity/aziot-identityd/src/lib.rs

```diff
@@ -28,11 +28,25 @@ const DPS_ASSIGNMENT_RETRY_INTERVAL_SECS: u64 = 10;
 /// This is the number of seconds to wait for DPS to complete assignment to a hub
 const DPS_ASSIGNMENT_TIMEOUT_SECS: u64 = 120;
 
+/// Private key types when creating a new identity certificate.
+#[derive(Clone)]
+enum KeyType<'a> {
+    /// Key handle managed by keyd.
+    Handle(&'a String),
```

Comment:
There's rarely a good reason to take a const-reference to a `String` directly instead of using a `&str`.

With `&String`, you end up doing a double pointer indirection whenever you access the data: once to dereference the `&String` into a `String`, and then one more time to access the underlying byte array.

* * *

That said, in this case, I think you'd be better off switching to `Handle(String)` directly, and not messing with any lifetimes.

Reply:
Fixed.